packages/rpc-graphql/src/resolvers/__tests__/account-resolver-test.ts

```diff
@@ -102,6 +178,247 @@ describe('account resolver', () => {
+            // FIXME: Prefer async version of this timer runner. See https://github.com/jestjs/jest/issues/14549
+            await Promise.resolve();
+            jest.runAllTimers();
+            expect(rpc.getAccountInfo).toHaveBeenCalledTimes(2); // Fixed with batch loader
+            expect(rpc.getAccountInfo).toHaveBeenLastCalledWith('AyGCwnwxQMCqaU4ixReHt8h5W4dwmxU7eM3BEQBdWVca', {
+                encoding: 'jsonParsed',
+            });
+        });
+        it('will resolve inline fragments with `jsonParsed` and the proper encoding when data and `jsonParsed` fields are requested', async () => {
+            expect.assertions(3);
+            const source = /* GraphQL */ `
+                query testQuery {
+                    account(address: "AyGCwnwxQMCqaU4ixReHt8h5W4dwmxU7eM3BEQBdWVca") {
+                        ... on MintAccount {
+                            data(encoding: BASE_58)
+                            supply
+                        }
+                    }
+                }
+            `;
+            rpcGraphQL.query(source);
+            // FIXME: Prefer async version of this timer runner. See https://github.com/jestjs/jest/issues/14549
+            await Promise.resolve();
+            jest.runAllTimers();
+            expect(rpc.getAccountInfo).toHaveBeenCalledTimes(3); // Fixed with batch loader
+            expect(rpc.getAccountInfo).toHaveBeenCalledWith('AyGCwnwxQMCqaU4ixReHt8h5W4dwmxU7eM3BEQBdWVca', {
+                encoding: 'jsonParsed',
+            });
+            expect(rpc.getAccountInfo).toHaveBeenCalledWith('AyGCwnwxQMCqaU4ixReHt8h5W4dwmxU7eM3BEQBdWVca', {
+                encoding: 'base58',
+            });
+        });
+        it('will resolve inline fragments with only the proper encoding not `jsonParsed` when no `jsonParsed` fields are requested', async () => {
+            expect.assertions(3);
+            const source = /* GraphQL */ `
+                query testQuery {
+                    account(address: "AyGCwnwxQMCqaU4ixReHt8h5W4dwmxU7eM3BEQBdWVca") {
+                        ... on MintAccount {
+                            data(encoding: BASE_58)
+                            lamports
+                            space
+                        }
+                    }
+                }
+            `;
+            rpcGraphQL.query(source);
+            // FIXME: Prefer async version of this timer runner. See https://github.com/jestjs/jest/issues/14549
+            await Promise.resolve();
+            jest.runAllTimers();
+            expect(rpc.getAccountInfo).toHaveBeenCalledTimes(2); // Fixed with batch loader
+            expect(rpc.getAccountInfo).toHaveBeenCalledWith('AyGCwnwxQMCqaU4ixReHt8h5W4dwmxU7eM3BEQBdWVca', {
+                encoding: 'base58',
+            });
+            expect(rpc.getAccountInfo).not.toHaveBeenCalledWith('AyGCwnwxQMCqaU4ixReHt8h5W4dwmxU7eM3BEQBdWVca', {
+                encoding: 'jsonParsed',
+            });
```

Comment:
@steveluscher It seems to me that the resolver is working correctly. I'm wondering if, in regards to your comments, you may have overlooked the `.not` here?

Reply:
🪦 OMG I missed that. Nice work!